geonode_mapstore_client/client/js/api/geonode/v2/index.js

```diff
@@ -645,7 +645,10 @@ export const copyResource = (resource) => {
 };
 
 export const downloadResource = (resource) => {
-    const url = resource.download_url || resource.href;
+    const { url, ajaxSafe } = getDownloadUrlInfo(resource);
+    if (!ajaxSafe) {
+        return Promise.reject(new Error("Not ajax safe"));
+    }
```

Comment:
I thought the plugins handles all checks external link including the ajax safe check now. Maybe this is not needed I added it just in case. Kindly let me know if you also want me to handle it here